client/cli/src/params/offchain_worker_params.rs

```diff
@@ -0,0 +1,92 @@
+// Copyright 2018-2020 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+use structopt::StructOpt;
+use sc_service::{
+	Configuration, ChainSpecExtension, RuntimeGenesis,
+};
+
+use sc_service::config::OffchainWorkerConfig;
+use structopt::clap::arg_enum;
+
+
+use crate::error;
+
+arg_enum! {
+	/// Whether off-chain workers are enabled.
+	#[allow(missing_docs)]
+	#[derive(Debug, Clone)]
+	pub enum OffchainWorkerEnabled {
+		Always,
+		Never,
+		WhenValidating,
+	}
+}
+
+/// Offchain worker related parameters.
+#[derive(Debug, StructOpt, Clone)]
+pub struct OffchainWorkerParams {
+
+	/// Should execute offchain workers on every block.
+	///
+	/// By default it's only enabled for nodes that are authoring new blocks.
+	#[structopt(
+		long = "offchain-worker",
+		value_name = "ENABLED",
+		possible_values = &OffchainWorkerEnabled::variants(),
+		case_insensitive = true,
+		default_value = "WhenValidating"
+    )]
+    pub enabled: OffchainWorkerEnabled,
+
+	/// Allow writing from the runtime to the offchain worker database directly (buffered).
```

Comment:
I'd call this: "Offchain workers indexing API". And emhpasize that if enabled, block import can write (buffered) data directly to offchain worker's DB.

Reply:
Naming is hard. I am open to suggestions for the individual items, it feels like all of them get very lengthy (especially when taking windows cmdlines into consideration)